### PR TITLE
feat: unify shell apps

### PR DESCRIPTION
**Pull request by znikola**
This PR lays a foundation for one shell app to rule them all.

### CODE REVIEW

File: projects/storefrontapp/src/app/configs/app-config.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    },
+  },
+  // we bring in static translations to be up and running soon right away
+  i18n: {
+    resources: translations,
+    chunks: translationChunksConfig,
+    fallbackLang: 'en',
+  },
+  features: {
+    level: '1.5',
```
Comment (marlass) [inline projects/storefrontapp/src/app/configs/app-config.ts]: Can we also bump it to 2.0?

File: angular.json [hunk 2]
```diff
@@ -114,9 +126,156 @@
...
+                  "maximumWarning": "6kb"
+                }
+              ],
+              "fileReplacements": [
+                {
+                  "replace": "projects/storefrontapp/src/environments/cds/cds.environment.ts",
+                  "with": "projects/storefrontapp/src/environments/cds/cds.environment.dev.ts"
+                },
+                {
+                  "replace": "projects/storefrontapp/src/environments/occ/occ.environment.ts",
```
Comment (marlass) [inline angular.json]: Shouldn't we align with normal app that in development we use the occ.environment.ts file?

File: angular.json [hunk 3]
```diff
@@ -114,9 +126,156 @@
...
+                }
+              ],
+              "fileReplacements": [
+                {
+                  "replace": "projects/storefrontapp/src/environments/cds/cds.environment.ts",
+                  "with": "projects/storefrontapp/src/environments/cds/cds.environment.dev.ts"
+                },
+                {
+                  "replace": "projects/storefrontapp/src/environments/environment.ts",
+                  "with": "projects/storefrontapp/src/environments/environment.prod.ts"
```
Comment (marlass) [inline angular.json]: And similarly in production we should use occ-1905 environemnts

File: angular.json [hunk 4]
```diff
@@ -210,6 +384,28 @@
...
+            "main": "projects/storefrontapp/server.ts",
+            "tsConfig": "projects/storefrontapp/tsconfig.server.json"
+          },
+          "configurations": {
+            "production": {
+              "outputHashing": "media",
+              "fileReplacements": [
+                {
+                  "replace": "projects/storefrontapp/src/environments/environment.ts",
+                  "with": "projects/storefrontapp/src/environments/cds/cds.environment.prod.ts"
```
Comment (marlass) [inline angular.json]: You should replace environment.ts with environment.prod.ts to have idential behavior with normal production build and the second replacement for cds.environment.ts should be to cds.environment.dev.ts. We don't have production configuration.
Additionally inclue occ-environment to be replaced with 1905 version like in normal production build (also applies to normal server)

File: angular.json [hunk 5]
```diff
@@ -223,6 +419,25 @@
...
+            "browserTarget": "storefrontapp:build:cds.development",
+            "serverTarget": "storefrontapp:server-cds"
+          },
+          "configurations": {
+            "production": {
+              "browserTarget": "storefrontapp:build:cds.production",
+              "serverTarget": "storefrontapp:server-cds:production",
+              "fileReplacements": [
+                {
+                  "replace": "projects/storefrontapp/src/environments/environment.ts",
```
Comment (marlass) [inline angular.json]: Do we need replacement in this particular config?

**Review by marlass — CHANGES_REQUESTED** [review]

File: angular.json (hunk 5 above)
Comment (znikola) [reply angular.json]: Nope, it's a leftover. Thanks for catching it.

File: angular.json (hunk 4 above)
Comment (znikola) [reply angular.json]: Thanks for pointing it out, done.

File: angular.json (hunk 3 above)
Comment (znikola) [reply angular.json]: Good catch.

File: angular.json (hunk 2 above)
Comment (znikola) [reply angular.json]: We could do that. 
I was just thinking to make it more convenient to start CDS (in local) and to point to dev17 (as it's already configured).

**Review by znikola — COMMENTED** [review]
Thanks for the review @marlasak .
We have to figure out how to solve the following:
- run `yarn build:core:lib && yarn build`
- we get 
```shell
ERROR in projects/storefrontapp/src/app/configs/cds-config.ts:1:27 - error TS2307: Cannot find module '@spartacus/cds'.

1 import { CdsConfig } from '@spartacus/cds';
                            ~~~~~~~~~~~~~~~~
projects/storefrontapp/src/app/app.module.ts:11:27 - error TS2307: Cannot find module '@spartacus/cds'.

11 import { CdsModule } from '@spartacus/cds';
```

One solution would be to run `yarn build:core:lib:cds` on the CI instead of `yarn build:core:lib`. This will of course increase the build time a bit.

File: projects/storefrontapp/src/environments/models/feature.environment.model.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { ModuleWithProviders } from '@angular/core';
+
+export interface FeatureEnvironment {
+  enabled: boolean;
+  feature?: ModuleWithProviders;
```
Comment (Platonn) [inline projects/storefrontapp/src/environments/models/feature.environment.model.ts]: What when the module to be loaded is not necesarily a result of a static method (like forRoot()), but only a simple module class, will this typing work?

Comment (znikola) [reply projects/storefrontapp/src/environments/models/feature.environment.model.ts]: Good point, we can use `Type<any> | ModuleWithProviders<any>` in that case.

File: angular.json [hunk 7]
```diff
@@ -223,6 +386,20 @@
             }
           }
         },
+        "serve-ssr-cds": {
+          "builder": "@nguniversal/builders:ssr-dev-server",
+          "options": {
+            "browserTarget": "storefrontapp:build:cds.development",
+            "serverTarget": "storefrontapp:server-cds"
```
Comment (marlass) [inline angular.json]: In this case we also need fileReplacement for "default" mode of server-cds, not only in production.

File: angular.json [hunk 8]
```diff
@@ -210,6 +347,32 @@
             }
```
Comment (marlass) [inline angular.json:347]: I noticed that we still have some references to environment.prod.ts file.

File: package.json [hunk 9]
```diff
@@ -80,26 +80,27 @@
...
+    "start:cds:ci:2005": "ng serve -c=cds.ci.2005 --prod",
+    "start:cds:ssl": "ng serve -c=cds.development --ssl",
     "test:cds:lib": "ng test cds --code-coverage",
-    "dev:cds:ssr": "ng run storefrontapp-cds-integration:serve-ssr",
-    "serve:cds:ssr": "node dist/storefrontapp-cds-integration-server/main.js",
-    "build:cds:ssr": "ng run storefrontapp-cds-integration:server:production",
-    "prerender:cds": "ng run storefrontapp-cds-integration:prerender",
+    "dev:cds:ssr": "cross-env NODE_TLS_REJECT_UNAUTHORIZED=0 ng run storefrontapp:serve-ssr-cds",
+    "prerender:cds:dev": "cross-env NODE_TLS_REJECT_UNAUTHORIZED=0 ng run storefrontapp:prerender-cds --routesFile projects/storefrontapp/prerender.txt",
+    "serve:cds:ssr": "node dist/storefrontapp-server/main.js",
```
Comment (marlass) [inline package.json]: Maybe we should drop that as only the build must be cds?

File: package.json [hunk 10]
```diff
@@ -80,26 +80,27 @@
...
-    "start:cds:ssl": "ng serve storefrontapp-cds-integration --ssl",
-    "start:cds:pwa": "cd ./dist/storefrontapp-cds-integration/ && http-server -p 4200",
-    "test:cds": "ng test storefrontapp-cds-integration",
+    "e2e:cy:cds:start-open": "start-server-and-test start:cds:ci:1905 http-get://localhost:4200 e2e:cy:open",
+    "e2e:cy:cds:start-run": "start-server-and-test start:cds:ci:1905 http-get://localhost:4200 e2e:cy:run",
+    "e2e:cy:cds:start-run-smoke-ci": "start-server-and-test start:cds:ci:1905 http-get://localhost:4200 e2e:cy:run:smoke:ci",
+    "e2e:cy:cds:start-run-ci": "start-server-and-test start:cds:ci:1905 http-get://localhost:4200 e2e:cy:run:ci:all",
+    "start:cds": "ng serve -c=cds.development --proxy-config mock-server/proxy.backend.js",
+    "start:cds:ci:1905": "ng serve -c=cds.ci.1905 --prod",
+    "start:cds:ci:2005": "ng serve -c=cds.ci.2005 --prod",
```
Comment (marlass) [inline package.json:93]: Could we also add for now the ccv2 option?

Comment (marlass) [reply package.json:93]: And do we also pass here the --prod flags?

**Review by marlass — CHANGES_REQUESTED** [review]

File: angular.json (hunk 7 above)
Comment (marlass) [reply angular.json]: I thought you would only set here: "serverTarget": "storefrontapp:server-cds:cds.development" without adding extra configuration here.

Comment (znikola) [reply angular.json]: Seemed safer this way. Do you agree?

Comment (marlass) [reply angular.json]: I would avoid completely the `cds.development` configuration here and use set it in this `401` line. It was already missed in package.json. ;)

**Review by marlass — APPROVED** [review]

**Pull request merged**